src/app.js

```diff
@@ -157,6 +161,30 @@ var run = function () {
 		});
 	});
 
+	// ----------------- file upload -------------
+
+	$('.inputFile').on('change', function () {
+		var fileList = $('input.inputFile')[0].files;
+		for (var i = 0; i < fileList.length; i++) {
+			var SOL_CACHE_FILE = utils.fileKey(fileList[i].name);
+			(function (name, file) {
+				var fileReader = new FileReader();
+				fileReader.onload = function (e) {
+					if (!window.localStorage[name] || confirm('The file ' + utils.fileNameFromKey(name) + ' already exists! Would you like to overwrite it?')) {
+						window.localStorage[name] = e.target.result;
+						sessions[name] = null;
+						updateFiles();
+					}
+				}
+				fileReader.readAsText(file);
+			})(SOL_CACHE_FILE, fileList[i]);
+		}
+
+		$filesEl.animate({left: Math.max( (0 - activeFilePos() + (FILE_SCROLL_DELTA/2)), 0)+ "px"}, "slow", function(){
```

Comment:
Please check how the spacing is in the rest of the file. Should be something similar to:
`+		$filesEl.animate({ left: Math.max((0 - activeFilePos() + (FILE_SCROLL_DELTA / 2)), 0) + "px" }, "slow", function () {`

And all quotation marks should be single quotes (`'`).